python/ray/dag/dag_operation_future.py

```diff
@@ -0,0 +1,90 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, List, Optional, TypeVar, Union
+
+T = TypeVar("T")
+
+
+class DAGOperationFuture(ABC, Generic[T]):
+    @abstractmethod
+    def wait(self, waiter: Optional[Any] = None):
+        raise NotImplementedError
+
+
+class ReadyFuture(DAGOperationFuture):
+    def __init__(self, result):
+        self._result = result
+
+    def wait(self, waiter: Optional[Any] = None):
+        return self._result
+
+
+class ListFuture(DAGOperationFuture):
```

Comment:
Why do we need this?

Reply:
See https://github.com/ray-project/ray/pull/47586#discussion_r1797514992

Reply:
Removed.